src/main/kotlin/com/openlattice/assembler/AssemblerConnectionManager.kt

```diff
@@ -416,6 +406,26 @@ class AssemblerConnectionManager(
         }
     }
 
+    private fun dropAndCreateMaterializedView(
+            connection: Connection,
+            tableName: String,
+            entitySetId: UUID,
+            materializablePropertyTypes: Map<UUID, PropertyType>) {
+        val selectColumns = getSelectColumnsForMaterializedView(materializablePropertyTypes.values)
+                .joinToString(",")
+
+        val sql = "SELECT $selectColumns FROM $PRODUCTION_FOREIGN_SCHEMA.${entitySetIdTableName(entitySetId)}"
+
+        val dropMaterializedEntitySet = "DROP MATERIALIZED VIEW IF EXISTS $tableName"
+        val createMaterializedViewSql = "CREATE MATERIALIZED VIEW IF NOT EXISTS $tableName AS $sql"
+
```

Comment:
is the "IF NOT EXISTS" necessary in createMaterializedViewSql? if the table did exist before it gets dropped and no longer exists. if it didn't exist before, it still doesn't exist.

Reply:
Nope, nice catch!